crates/devices/virtio-vsock/src/packet.rs

```diff
@@ -583,27 +594,39 @@ impl<'a, B: BitmapSlice> VsockPacket<'a, B> {
             return Err(Error::DescriptorLengthTooSmall);
         }
 
-        // All RX descriptor chains should have a header and a data descriptor.
-        if !chain_head.has_next() {
-            return Err(Error::DescriptorChainTooShort);
-        }
-
         let header_slice = mem
             .get_slice(chain_head.addr(), PKT_HEADER_SIZE)
             .map_err(Error::InvalidMemoryAccess)?;
-        let data_desc = desc_chain.next().ok_or(Error::DescriptorChainTooShort)?;
 
-        if !data_desc.is_write_only() {
-            return Err(Error::UnexpectedReadOnlyDescriptor);
-        }
+        // Starting from Linux 6.2 the virtio-vsock driver can use a single descriptor for both
+        // header and data.
+        let data_slice = if chain_head.len() as usize > PKT_HEADER_SIZE {
```

Comment:
We typically want to compare stuff.len() to the PKT_HEADER_SIZE. The `len` function uses an atypical return type for a `len` function which makes these conversions required. Maybe we can open an issue to fix this separately if this is not on purpose.

CC: @lauralt

Reply:
I think what we wanted here was to be consistent with the spec (len is defined as le32 there). We can also update the method to return an usize, but I would be in favor of keeping it as defined in the spec.

Reply:
I don't have a strong opinion on that, but I agree that we should change it in another PR.

Reply:
Ok, it looks like it's on purpose like this, so I will resolve this.